chessboard/chessboard.go

```diff
@@ -0,0 +1,48 @@
+package chessboard
+
+type Rank []bool
+
+type Chessboard map[string]Rank
+
+// CountInRank returns how many squares are occupied in the chessboard,
+// within the given rank
+func CountInRank(cb Chessboard, rank string) (count int) {
+	for i := range cb[rank]{
+		if cb[rank][i]{count += 1}
```

Comment:
That's a really odd way to write it. At first it looked like an slice until I saw the if. In any case, why not to do something like:
```go
for _, square := range cb[rank] {
		if square {
```

Reply:
Not really odd, just different approach to addressing it which is functionally and, at this scale, computationally equivalent.

Reply:
What I meant is that it make it look like an slice literal initialization. It's more frequent to see it as 3 lines instead of this. Even the go formatter formats it in that way.

Reply:
Ohhh... well, I might have not formatted it as I was just plunking this out before our meeting ;)